app.rb

```diff
@@ -0,0 +1,131 @@
+      end
+    end
+  end
+
+  def list_people
+    if @people.empty?
+      puts 'There\'s no one here ATM!'
+    else
+      @people.each do |person|
+        puts "Name: #{person.name} Age: #{person.age} ID: #{person.id}"
+      end
+    end
+  end
+
+  def create_person
+    puts 'Are you a:'
+    puts '1 - Student'
+    puts '2 - Teacher'
+    person_input = gets.chomp.to_i
+
+    case person_input
+    when 1 then create_student
+    when 2 then create_teacher
+    else
+      raise 'Please enter a valid option, number 1 or 2'
+    end
+  end
+
+  def create_student
+    puts 'What is your name?'
+    name = gets.chomp
+    puts 'How old are you?'
+    age = gets.chomp.to_i
+    puts 'What grade are you?'
+    classroom = gets.chomp
+    puts 'Do you have your parents\' permission? [Y/N]'
+    parent_permission = gets.chomp
+    student = Student.new(name, age, classroom, parent_permission: parent_permission)
+    @people << student
+    puts 'Student successfully created'
+  end
+
+  def create_teacher
+    puts 'What is your name?'
+    name = gets.chomp
+    puts 'How old are you?'
+    age = gets.chomp.to_i
+    puts 'Enter your specialization'
+    specialization = gets.chomp
+    puts 'Do you have your parents\' permission? [Y/N]'
+    parent_permission = gets.chomp
+    teacher = Teacher.new(name, age, specialization, parent_permission)
+    @people << teacher
+    puts 'Teacher Successfuly created'
+  end
+
+  def create_rental
+    return puts 'There are no books in the library.' if @books.empty?
+    return puts 'There are no people in the system.' if @people.empty?
+
+    puts 'Select a book by its number:'
+    @books.each_with_index do |book, index|
+      puts "Number: #{index + 1} - Title: #{book.title}, Author: #{book.author}"
+    end
+    book_id_input = gets.chomp.to_i
+
+    return puts 'Invalid book selection.' if book_id_input < 1 || book_id_input > @books.size
+
+    book = @books[book_id_input - 1]
+
+    puts 'Select the person renting the book by their number:'
+    @people.each_with_index do |person, index|
+      puts "Number: #{index + 1} - Role: #{person.class.name}, Name: #{person.name}, ID: #{person.id}"
+    end
+    person_id_input = gets.chomp.to_i
+
+    return puts 'Invalid person selection.' if person_id_input < 1 || person_id_input > @people.size
+
+    person = @people[person_id_input - 1]
+
+    puts 'Enter the rental date [yyyy-mm-dd]:'
+    date = gets.chomp
+    rental = Rental.new(date, book, person)
+    @rentals << rental
+    puts 'The book has been successfully rented!'
+  end
+
+  # rubocop:enable Metrics/CyclomaticComplexity
+  def list_all_rentals(_person_id)
+    if @rentals.empty?
+      puts 'No books have been rented out!'
+    else
+      puts 'Enter person\'s id'
+      id = gets.chomp.to_i
+      @rentals.each do |rental|
+        puts "#{rental.book.title} by #{rental.book.author}, rented on #{rental.date}" if rental.person.id == id
+      end
+    end
+  end

```

Comment:
- [ ] I think it would be better to replicate the given example by removing a duplicated question on the person's id while ``` Listing all rentals for a given person's id``` 🙏 

You may kindly check on this short record that I made for your reference 👇 
https://www.loom.com/share/6f40d419de934d2c84b92bf32f0a73bb